tempesta_fw/t/unit/test_http_parser.c

```diff
@@ -365,13 +483,53 @@ TEST(http_parser, parses_connection_value)
 		EXPECT_EQ(req->flags & __TFW_HTTP_CONN_MASK, TFW_HTTP_CONN_CLOSE);
 }
 
+TEST(http_parser, fuzzer)
+{
+	char *str;
+	int i = 0, ret;
+	str = vmalloc(2 * 1024 * 1024);
+
+	for (i = 0; i < 2; i++) {
+		TEST_LOG("fuzz request i: %d\n", i);
+		ret = fuzz_gen(str, 5, FUZZ_REQ);
+		switch (ret) {
+		case FUZZ_VALID:
+			FOR_REQ(str);
+			break;
+		case FUZZ_INVALID:
+			EXPECT_BLOCK_REQ(str);
+			break;
+		case FUZZ_END:
+			goto end;
```

Comment:
-EINVAL returned from fuzz_gen() is not handled in this switch and swith at the below.